### ISSUE TITLE
Possible to configure "domain" to value with periods?

### ISSUE THREAD

**Issue by petemounce** [hashicorp/consul#657]
I'm trying to set up consul in AWS, so that Route53 delegates to consul for entries that consul serves. Because Windows doesn't have an equivalent to dnsmasq, it strikes me as a bit risky to add `127.0.0.1` as a DNS server on each of my nodes (I don't know how Windows chooses which DNS server to use, and so on).

So, I have [set up bind on my consul servers to forward requests on port 53 to consul on port 8600](https://consul.io/docs/guides/forwarding.html) - that works when I `dig` locally - I can do `dig @localhost consul.service.consul` and get back the correct response. 

(I would love to know how to do that by just adding an iptables port-forward, to avoid needing to run bind, but that can wait).

Now what I want to do is create a hosted zone in Route53, linked to my VPC, so that I can get [Route53's plus-2 DNS server](http://docs.aws.amazon.com/Route53/latest/DeveloperGuide/hosted-zones-private.html) to delegate to consul.

I have set up my internal hosted zone `foowibble`; AWS Route53 automatically creates NS and SOA records for me, and I have left these untouched.

I've then created an NS record, `consul.tenant1.foowibble` whose value is
```
ns1.tenant1.foowibble
ns2.tenant1.foowibble
ns3.tenant1.foowibble
```
(I'm aiming at a multi-tenant setup, with one consul cluster per tenant.)

I've then launched my cluster, and for now, manually created 1 A record per server in my 3-node cluster.

My server nodes are configured to use `consul.tenant1.foowibble` as the `domain` via `consul.json` - but now, on any of them (and after `consul reload`), I cannot get 
```shell
dig @localhost consul.service.consul.tenant1.foowibble -p 8600
```
to respond, and so I certainly won't be able to get responses from the Route53 "plus 2" DNS.

Should I _expect_ consul to work with a domain with periods in it...?

If not, could it be made to? If it couldn't, what alternatives might I have to make this work...?

**Comment by petemounce** [comment]
I've done some further testing with 0.4.1, and I haven't been able to configure a value for `domain` to be anything different from the default.

I've tried, in `consul.json`
* `"domain": "wibble"`
* `"domain": "wibble."`
and I restarted the consul process as well as used `consul reload`.

I expected to see
* `consul monitor` output errors about mis-configuration (if I'd made a mistake)
* `dig @127.0.0.1 -p 8600 consul.service.wibble` to give an `ANSWER` with my IP (I'm running consul on my workstation in `-bootstrap-expect 1` mode.

As an aside, it would be nice if consul-ui would include DNS names for services and nodes.
